### PR TITLE
Added Typescript definition generation.

### PR DESCRIPTION
**Pull request by masterlink950**
Added Typescript definition generation

### CODE REVIEW

File: scripts/generate.js [hunk 1]
```diff
@@ -44,6 +53,43 @@ for (let component of components) {
   fs.writeFileSync(`${__dirname}/../build/${component.name}Icon.js`, fileContent);
 
   indexContent += `export ${component.name} from './${component.name}Icon';\n`;
+  
+  /*
+    Pushing the different formats of typescript definitions to their respective arrays.
+  */
+  defFileFillIns.declareIconClasses.push(`class ${component.name}Icon extends GenericMdiIcon {}`);
```
Comment (levrik) [inline scripts/generate.js]: Why not `extends MdiIcon {}` and `class MdiIcon extends __React.Component<IMdiIconProps, {}> {}` in line 84?

Comment (masterlink950) [inline scripts/generate.js]: I couldn't remember why, but there was a specific reason why it was done this way. So I tried to modify it and I immediately get errors.

**Comment by levrik** [comment]
@masterlink950 Sorry for the delay. I will still add declarations for flow and then release these two together. Doing this next weekend.

**Comment by levrik** [comment]
@masterlink950 Could you please resolve the merge conflicts?

**Comment by masterlink950** [comment]
Sorry it took me so long to get back to this. Been busy with other projects. We should be good now though.

File: README.md [hunk 2]
```diff
@@ -28,3 +28,15 @@ const MyComponent = () => {
 ```
 
 The icons get a class named `mdi-icon` attached for styling. You can also attach own additional classes with the `className` property.
+
+*Note that the second method from the example imports every icon from this package. Use the first method if your bundler does not support the removal of unused code.*
```
Comment (levrik) [inline README.md]: This needs to be removed.

Comment (masterlink950) [inline README.md]: Finished. I'm not sure how (or if I can) to mark this change request as completed though.

**Comment by levrik** [comment]
@masterlink950 Here there, I decided to not ship TypeScript definitions for now because I don't use TypeScript and cannot test them out.

**Comment by masterlink950** [comment]
No problem. That's understandable.

**Pull request closed without merging**